plugin/trino-hive/src/main/java/io/trino/plugin/hive/projection/InjectedProjection.java

```diff
@@ -43,11 +44,12 @@ public List<String> getProjectedValues(Optional<Domain> partitionValueFilter)
         Domain domain = partitionValueFilter
                 .orElseThrow(() -> new InvalidProjectionException(columnName, "Injected projection requires single predicate for it's column in where clause"));
         Type type = domain.getType();
-        if (!domain.isNullableSingleValue() || !canConvertSqlTypeToStringForParts(type, true)) {
+        ValueSet values = domain.getValues();
+        if (!values.isDiscreteSet() || !canConvertSqlTypeToStringForParts(type, true)) {
             throw new InvalidProjectionException(columnName, "Injected projection requires single predicate for it's column in where clause. Currently provided can't be converted to single partition.");
         }
-        return Optional.ofNullable(sqlScalarToString(type, domain.getNullableSingleValue(), null))
-                .map(ImmutableList::of)
-                .orElseThrow(() -> new InvalidProjectionException(columnName, type));
+        return values.getDiscreteSet().stream()
+                .map(value -> sqlScalarToString(type, value, null))
```

Comment:
The previous iteration handled the scenario where `sqlScalarToString(type, value, null)` returned `null` by throwing `InvalidProjectionException`. If that were to occur now, it would blow up inside of the `toImmutableList()` collector. That said, the situation should be impossible- because it appears to only return `null` for `null` inputs but for completeness / defensiveness we probably want to include a specific check on the return to throw the appropriate exception (our code does the same thing for the same reason).

Reply:
Makes sense, thanks for sharing!
I've updated the code to throw an exception using the previous logic and message.
Let me know if you'd prefer a more specific or customized error message, and also if you're okay with us always reporting the first value that returns null